### PR TITLE
Split integration tests by domain

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary
- replace the 5,960-line integration test monolith with eight cohesive collected modules
- keep shared device, session, feed, and text-case helpers in a non-collected support module
- preserve collection, markers, parametrization, and active CI selection while updating commands and docs

## Validation
- exact 134-item normalized collection/marker/parameter equivalence
- exact 25-item fast-integration CI selector equivalence
- isolated collection for every split module
- 21 network-free/small-config CPU integration cases
- 7 passed and 4 existing skips across representative real-checkpoint CPU cases
- split `build_graph_*` and non-integration test suite
- repository lint/format
- independent GPT-5.6 Sol review

Stacked on #704 (`justinchuby-split-l1-graph-tests`).

### CODE REVIEW

File: tests/integration/generation_test.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+@pytest.mark.integration_fast
+@pytest.mark.parametrize("model_id,trust_remote_code", _TEXT_MODELS)
+class TestGreedyGeneration:
+    """Compare greedy text generation between ONNX and PyTorch."""
+
+    def test_generate_tokens_match(self, model_id: str, trust_remote_code: bool):
+        """Generated token IDs should be identical for greedy decoding."""
+        onnx_model = build(model_id, dtype="f32", load_weights=True)
+        torch_model, tokenizer = load_torch_model(model_id)
+        config = _get_config(model_id, trust_remote_code)
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/integration/generation_test.py:38]: Keeping this unchanged intentionally. The generation suite preserves `main`'s original loader/build calls exactly, and `trust_remote_code` continues to feed `_get_config`. Passing it into additional loaders would change test behavior, while this PR's acceptance criterion is exact collection/parameter metadata and behavior preservation. That functional cleanup should be handled separately.

File: tests/integration/vlm_test.py [hunk 2]
```diff
@@ -0,0 +1,1254 @@
...
+    ):
+        """Prefill with a short prompt, no image inputs."""
+        module_class = getattr(models, module_class_name)
+        onnx_model = build(
+            model_id,
+            module_class=module_class,
+            task="text-generation",
+            dtype="f32",
+            load_weights=True,
+        )
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/integration/vlm_test.py:291]: Keeping this unchanged intentionally. This `build(...)` invocation is the original `main` behavior; the parameter remains consumed by `_get_config(model_id, trust_remote_code)`. Passing it to `build` would be a functional change rather than an organization-only move, so this PR preserves the existing behavior and exact parametrization.

File: tests/integration/text_test.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+@pytest.mark.integration_fast
+@pytest.mark.parametrize("model_id,trust_remote_code", _TEXT_MODELS)
+class TestForwardNumerical:
+    """Compare single forward pass logits between ONNX and PyTorch."""
+
+    def test_prefill_logits_match(self, model_id: str, trust_remote_code: bool):
+        """First forward pass (prefill) with a short prompt."""
+        onnx_model = build(model_id, dtype="f32", load_weights=True)
+        torch_model, tokenizer = load_torch_model(model_id)
+        config = _get_config(model_id, trust_remote_code)
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/integration/text_test.py:38]: Keeping this unchanged intentionally. These `build(...)` and `load_torch_model(...)` calls are byte-for-byte the behavior from `main`'s original `tests/integration_test.py`; this PR is restricted to test organization. The parameter is still used by `_get_config(model_id, trust_remote_code)`, and threading it into the runtime loaders would be a functional test change outside this refactor. Exact parameter metadata and execution behavior remain preserved.

File: tests/integration/encoder_media_seq2seq_test.py [hunk 4]
```diff
@@ -0,0 +1,755 @@
...
+
+    def test_hidden_states_match(self, model_id: str, trust_remote_code: bool):
+        """Forward pass: input_ids → last_hidden_state."""
+        from mobius._testing.torch_reference import (
+            load_torch_encoder_model,
+            torch_encoder_forward,
+        )
+
+        onnx_model = build(model_id, dtype="f32", load_weights=True)
+        torch_model, tokenizer = load_torch_encoder_model(model_id)
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/integration/encoder_media_seq2seq_test.py:74]: Keeping the parameter shape unchanged intentionally because this PR must preserve all 134 exact parameter records and runtime behavior from `main`'s monolith. The encoder call is unchanged from the original suite; removing the parameter or threading it into loaders would be a functional test change outside this organization refactor.

File: tests/integration/_support.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+        from huggingface_hub import model_info
+
+        model_info(model_id)
+    except Exception:
+        return False
+    else:
+        return True
+
+
+_TEXT_MODELS = [
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/integration/_support.py]: Fixed in 82694156. The catalogue is now the public cross-module support symbol `TEXT_MODELS`, and both collected consumers plus contributor/skill references were updated. This keeps the exact parametrization unchanged while making the intended external use explicit.
